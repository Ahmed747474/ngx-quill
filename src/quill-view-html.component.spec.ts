import { Component, ViewChild } from '@angular/core'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'

import { QuillViewHTMLComponent } from '../src/quill-view-html.component'

import { QuillModule } from './quill.module'

describe('Basic QuillViewHTMLComponent', () => {
  let fixture: ComponentFixture<QuillViewHTMLComponent>

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        QuillModule.forRoot()
      ]
    })

    fixture = TestBed.createComponent(QuillViewHTMLComponent)
  })

  it('should render and set default snow theme class', async(async () => {
    const element = fixture.nativeElement
    fixture.detectChanges()
    await fixture.whenStable()

    expect(element.querySelectorAll('.ql-editor').length).toBe(1)
    expect(fixture.componentInstance.themeClass).toBe('ql-snow')
    const viewElement = element.querySelector('.ql-container.ql-snow > .ql-editor.ngx-quill-view-html')
    expect(viewElement).toBeDefined()
  }))
})

describe('QuillViewHTMLComponent - content', () => {
  @Component({
    template: `
  <quill-view-html [content]="content" theme="snow"></quill-view-html>
  `
  })
  class HTMLComponent {
    @ViewChild(QuillViewHTMLComponent, {
      static: true
    }) view: QuillViewHTMLComponent | undefined
    content = '<p>Hallo</p>'
  }

  let fixture: ComponentFixture<HTMLComponent>

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [HTMLComponent],
      imports: [QuillModule]
    })

    fixture = TestBed.createComponent(HTMLComponent) as ComponentFixture<HTMLComponent>
    fixture.detectChanges()
  })
  it('should be set html', async(async () => {
    const element = fixture.nativeElement

    await fixture.whenStable()
    const viewElement = element.querySelector('.ql-container.ql-snow > .ql-editor.ngx-quill-view-html')
    expect(viewElement.innerHTML).toEqual('<p>Hallo</p>')
  }))

  it('should update html', async(async () => {
    const component = fixture.componentInstance
    await fixture.whenStable()
    component.content = '<p>test</p>'
    fixture.detectChanges()
    await fixture.whenStable()

    const element = fixture.nativeElement
    const viewElement = element.querySelector('.ql-container.ql-snow > .ql-editor.ngx-quill-view-html')
    expect(viewElement.innerHTML).toEqual('<p>test</p>')
  }))
})
