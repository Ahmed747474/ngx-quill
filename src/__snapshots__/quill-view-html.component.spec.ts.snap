// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic QuillViewHTMLComponent should render and set default snow theme class 1`] = `
<quill-view-html
  config={[Function Object]}
  content=""
  innerHTML=""
  sanitizer={[Function DomSanitizerImpl]}
  themeClass={[Function String]}
>
  <div
    class="ql-container ql-snow"
  >
    <div
      class="ql-editor"
    />
  </div>
</quill-view-html>
`;

exports[`QuillViewHTMLComponent - content should be set html 1`] = `
<ng-component
  content={[Function String]}
  view={[Function QuillViewHTMLComponent]}
>
  <quill-view-html
    theme="snow"
  >
    <div
      class="ql-container ql-snow ngx-quill-view-html"
    >
      <div
        class="ql-editor"
      >
        <p>
          Hallo
        </p>
      </div>
    </div>
  </quill-view-html>
</ng-component>
`;

exports[`QuillViewHTMLComponent - content should update html 1`] = `
<ng-component
  content={[Function String]}
  view={[Function QuillViewHTMLComponent]}
>
  <quill-view-html
    theme="snow"
  >
    <div
      class="ql-container ql-snow ngx-quill-view-html"
    >
      <div
        class="ql-editor"
      >
        <p>
          test
        </p>
      </div>
    </div>
  </quill-view-html>
</ng-component>
`;
